x-pack/plugins/security_solution/server/endpoint/routes/metadata/query_builders.ts

```diff
@@ -9,6 +9,7 @@ import { EndpointAppContext } from '../../types';
 
 export interface QueryBuilderOptions {
   unenrolledAgentIds?: string[];
+  statusAgentIDs?: string[] | null;
```

Comment:
why the null? the `?` means it is optional

Reply:
optional is only satisfied when the value is potentially `undefined`. It does not handle the case when the value is `null`. 

`null` was the value I was using to indicate not-passed-in, vs undefined handling the case of invalid things passed in. Since we have the type checker validating that now, I can switch the unused case to be `undefined` now.

But just pointing out, `null` does not work with `?` optionals.